packages/react-dom/src/__tests__/ReactDOMInput-test.js

```diff
@@ -167,6 +167,57 @@ describe('ReactDOMInput', () => {
     document.body.removeChild(container);
   });
 
+  it('should handle when initial state is 0', () => {
+    class App extends React.Component {
+      inputRef = null;
+      constructor(props) {
+        super(props);
+        this.state = {
+          value: 0,
+        };
+      }
```

Comment:
Since you have a class field for `inputRef`, what do you think about assigning state this way too? Like:

```
state = {
  value: 0,
}
```